Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -540,12 +539,77 @@ theorem cliqueSet_map_of_equiv (G : SimpleGraph α) (e : α ≃ β) (n : ℕ) :
 
 end CliqueSet
 
+/-! ### Clique number -/
+
+
+section CliqueNumber
+
+variable {α : Type*} {G : SimpleGraph α}
+
+/-- The maximal number of vertices in a graph `G`. -/
+noncomputable def cliqueNum (G : SimpleGraph α) : ℕ := sSup {n | ∃ s, G.IsNClique n s}
+
+/-- A maximum clique in a graph `G` is a clique with the largest possible size. -/
+structure IsMaximumClique (G : SimpleGraph α) (s : Finset α) : Prop where
+  (isClique : G.IsClique s)
+  (maximum : ∀ t : Finset α, G.IsClique t → #t ≤ #s)
+
+theorem isMaximumClique_iff {s : Finset α} :
+    G.IsMaximumClique s ↔ G.IsClique s ∧ ∀ t : Finset α, G.IsClique t → #t ≤ #s :=
+  ⟨fun h ↦ ⟨h.1, h.2⟩, fun h ↦ ⟨h.1, h.2⟩⟩
+
+/-- A maximal clique in a graph `G` is a clique that cannot be extended by adding more vertices. -/
+structure IsMaximalClique (G : SimpleGraph α) (s : Finset α) : Prop where
+  (clique : G.IsClique s)
+  (maximal : ∀ t : Finset α, G.IsClique t → s ⊆ t → t = s)
+
+theorem isMaximalClique_iff {s : Finset α} :
+    G.IsMaximalClique s ↔ G.IsClique s ∧ ∀ t : Finset α, G.IsClique t → s ⊆ t → t = s :=
+  ⟨fun h ↦ ⟨h.1, h.2⟩, fun h ↦ ⟨h.1, h.2⟩⟩
+
+lemma maximal_of_maximum (s : Finset α) (M : G.IsMaximumClique s) : G.IsMaximalClique s :=
```

Comment:
```suggestion
lemma IsMaximumClique.isMaximalClique (s : Finset α) (M : G.IsMaximumClique s) : G.IsMaximalClique s :=
```
(provided I haven't got something wrong), this will mean you can write `M.isMaximalClique` in applications